src/EventSubscriber/XLocationIdResponseSubscriber.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformHttpCacheBundle\EventSubscriber;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Rewrites the X-Location-Id HTTP header.
+ *
+ * This is a BC layer for custom controllers (including REST server) still
+ * using X-Location-Id header which is now deprecated. For
+ * full value of tagging, see docs/using_tags.md for how to take advantage of the
+ * system.
+ */
+class XLocationIdResponseSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $tagHeader;
+
+    /**
+     * @var LoggerInterface|null
+     */
+    private $logger;
+
+    public function __construct($tagHeader, LoggerInterface $logger = null)
+    {
+        $this->tagHeader = $tagHeader;
+        $this->logger = $logger;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [KernelEvents::RESPONSE => ['rewriteCacheHeader', -5]];
+    }
+
+    public function rewriteCacheHeader(FilterResponseEvent $event)
+    {
+        $response = $event->getResponse();
+        if (!$response->headers->has('X-Location-Id')) {
+            return;
+        }
+
+        if ($this->logger instanceof LoggerInterface) {
+            $this->logger->notice(
+                'X-Location-Id rewritten to use ezplatform-http-cache tag header, see \<bundle\>/docs/using_tags.md for more info'
+            );
+        }
+
+        // Map the tag, and even if it was not officially supported, make sure to handle comma separated values as was possible with Varnish
+        $tags = array_map(
+            function ($id) {return 'location-' . trim($id);},
+            explode(',', $response->headers->get('X-Location-Id'))
+        );
+
+        // Check if there is already exiting tags and merge them
+        if ($response->headers->has($this->tagHeader)) {
+            $tags += $response->headers->get($this->tagHeader, null, false);
+        }
+
+        // Set headers
```

Comment:
I know this is WIP but I really wouldn't leave such comments :-)

Also applies to the comments above. If the statements aren't readable (the `->has()` one is readable, in my opinion, for instance), private functions can be used.